node/core/chain-selection/src/tree.rs

```diff
@@ -394,6 +394,47 @@ fn apply_reversions(
 	Ok(())
 }
 
+/// Marks a single block as explicitly reverted, then propagates viability updates
+/// to all its children. This is triggered when the disputes subsystem signals that
+/// a dispute has concluded against a candidate.
+pub(crate) fn apply_single_reversion(
+	backend: &mut OverlayedBackend<impl Backend>,
+	revert_hash: Hash,
+	revert_number: BlockNumber,
+) -> Result<(), Error> {
+	let mut entry_to_revert = match backend.load_block_entry(&revert_hash)? {
```

Comment:
I factored out the shared code into a helper instead, since I think combining the two directly could make the code quite a bit harder to interpret. Take a look when you get the chance please!

Reply:
Looks good, thank you!